15314f1ddb882700dc9fabc5ca961943/update/sys_script_include_b630a09ddb9020d01206fd641d9619df.xml

```diff
@@ -718,20 +718,355 @@ IllumioUtils.prototype = {
+				if (responseDict && !responseDict["status"]) {
+					gs.error("Failed fetching of Label Dimensions from Illumio PCE for PCE configuration " + pceConfig.getValue("name") + ".");
+					this.persistLabelDimensions(pceConfig);
+					continue;
+				}
+				var response = responseDict["response"];
+				response_status = responseDict["statusCode"];
+
+				if (response_status >= 200 && response_status < 300) {
+					response = JSON.parse(response.getBody());
+					
+					// Iterate over each dimension and intialize record, if not already present, and update it.
+					for (var i = 0; i < response.length; i++) {
+						this.createOrUpdateLabelDimensionRecord(response[i], pceConfig);
+					}
+					gs.debug("Successfully fetched " + response.length + " label dimensions from Illumio PCE for PCE configuration " + pceConfig.getValue("name")) + ".";
+				} else {
+					gs.error("Error while fetching label dimensions from Illumio PCE for PCE configuration " + pceConfig.getValue("name") + ". Status code - " + response_status + ". Response - " + JSON.stringify(response));
+					this.persistLabelDimensions(pceConfig);
+				}
+			}
+        } catch (exc) {
+            gs.error("Exception while fetching label dimensions from PCE. " + exc);
+        }
+    },
+	
+	/**
+	 * Fetch all label dimensions and check if a label dimension with same key is already present on Illumio PCE
+	 * @param {String} key for the newly created label dimension
+	 * @param {String} pceConfig sys_id of the configuration record
+	 * @returns {Boolean} status of whether the label dimension has to be created
+	 */
+	validateLabelDimensionCreation: function (key, pceConfigSysId) {
+		try {
+			var matchFound = false;
+			
+			var pceConfig = new GlideRecord("x_illu2_illumio_illumio_pce_conf");
+			if(pceConfig.get(pceConfigSysId)) {
+				var responseDict = this.getLabelDimensions(pceConfig);
+				if (!responseDict["status"]) {
+					return matchFound;
+				}
+				response = responseDict["response"];
+				response_status = responseDict["statusCode"];
+
+				if (response_status >= 200 && response_status < 300) {
+					response = JSON.parse(response.getBody());
+
+					// Iterate over each dimension and intialize record, if not already present, and update it.
+					for (var i = 0; i < response.length; i++) {
+						this.createOrUpdateLabelDimensionRecord(response[i], pceConfig);
+
+						if (response[i]["key"] == key) {
+							matchFound = true;
+							gs.debug("A label dimension with key " + key + " is already present on Illumio PCE for PCE configuration " + pceConfig.getValue("name") + ". Replacing the label dimension details with the ones obtained from Illumio PCE.");
+						}
+					}
+				} else {
+					gs.error("Error while validating the creation of label dimension with key - " + key + " for the PCE configuration " + pceConfig.getValue("name") + ". Status code - " + response_status + ". Response - " + JSON.stringify(response));
+					return false;
```

Comment:
it is ideal to set `matchfound` to false and return that at the end of the method

Reply:
Noted! I’ll update this with branch - “n-arbitrary-label-dimensions”